### PR TITLE
Update where app-model actions/providers/processors live in codebase 

### PR DESCRIPTION
**Pull request by lucyleeow**
# References and relevant issues
Relates to https://github.com/napari/napari/pull/6848
Depends on https://github.com/napari/napari/pull/6743

# Description

Updates the info on where actions live in the napari codebase.

### CODE REVIEW

**Comment by lucyleeow** [comment]
cc @DragaDoncila

File: docs/developers/architecture/app_model.md
Comment (DragaDoncila) [inline docs/developers/architecture/app_model.md]:
```suggestion
of living in a menu and are thus considered GUI actions. If we move to supporting more GUI backends, we may want to move these out of `_qt` to a shared `gui` folder, depending on implementation.
```

File: docs/developers/architecture/app_model.md
Comment (DragaDoncila) [inline docs/developers/architecture/app_model.md]:
```suggestion
This also ensures that there is only one file defining both actions and their submenus.
```

File: docs/developers/architecture/app_model.md
Comment (DragaDoncila) [inline docs/developers/architecture/app_model.md]:
```suggestion
The layer context menu actions do not require a GUI (they only require the `layerlist`) and thus live in
```

**Review by DragaDoncila — COMMENTED** [review]
Thanks @lucyleeow just some nits here

**Comment by lucyleeow** [comment]
Thanks, changes made!

**Review by DragaDoncila — APPROVED** [review]
Awesome sauce, thanks @lucyleeow!

**Comment by lucyleeow** [comment]
AH whoops sorry, I forgot this has already been approved.

I realise that we now no longer have qt providers or processors anymore, so I've updated that here too. Sorry!

**Comment by DragaDoncila** [comment]
@lucyleeow do we now have to wait for #6743 to go in before we merge this?

**Comment by lucyleeow** [comment]
:facepalm: yes you are right. I'll add a depends on

**Comment by Czaki** [comment]
test comment

**Comment by lucyleeow** [comment]
The depends on has now been merged (not sure why the CI is still failing) so maybe this could go in now, @DragaDoncila ?

**Comment by DragaDoncila** [comment]
thanks @lucyleeow sorry for the delay...

**Pull request merged**
